sympy/core/tests/test_evalf.py

```diff
@@ -236,6 +236,7 @@ def test_evalf_bugs():
 
 
 def test_evalf_integer_parts():
+    from sympy.functions.elementary.trigonometric import tan
```

Comment:
Could we move this up to the top of the file, maybe to lines 2 or 3?

Reply:
I've been wanting to ask what convention is followed with imports. Since `tan` is needed only in this function and nowhere else in this file, I think the sympy codebase prefers keeping the import in the function itself (see also `test_implemented_function_evalf` and `test_issue_5486_bug` in the same file).

But in what cases is the import preferred to be kept at the start of the file? Perhaps when more than 1 function needs it (reduce multiple lines of code to 1)? Or perhaps importing in the function or at the start of the file not make much of a difference?

Reply:
Delayed imports are used to solve problems with cyclic imports. That shouldn't be needed in test files.